### PR TITLE
Systemd's Watchdog bee

### PR DESCRIPTION
**Pull request by rubiojr**
Beehive integration with Systemd's watchdog.

Linux only.

http://0pointer.de/blog/projects/watchdog.html

### CODE REVIEW

File: bees/watchdogbee/watchdogbee.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	for {
+		select {
+		case <-mod.SigChan:
+			return
+		case <-time.After(time.Duration(interval) * time.Second):
+			resp, err := http.Get(api.CanonicalURL().String())
+			if err == nil {
+				resp.Body.Close()
+				mod.LogDebugf("Notify Systemd's watchdog")
+				daemon.SdNotify(false, "WATCHDOG=1")
```
Comment (muesli) [inline bees/watchdogbee/watchdogbee.go]: Coincidentally I fixed upstream `go-systemd` a year ago to not rely on those magic-strings anymore:

https://github.com/coreos/go-systemd/commit/39ad653e9c19ed64c183ee84c876e5ff37ad7a49#diff-1f15fcf5421eecd6ca1b61337c319495

So we can simple use:

```
daemon.SdNotify(false, SdNotifyWatchdog)
```

**Comment by muesli** [comment]
Thinking about this a bit more, maybe this should rather be core functionality instead of a bee?

**Comment by rubiojr** [comment]
> Thinking about this a bit more, maybe this should rather be core functionality instead of a bee?

Yeah, I thought about it, but given that multi OS support for this would be challenging I opted for this approach.

If you'd like to see this integrated into the core, conditionally enabling it when systemd is detected, I'd be happy to give it a try, it'd definitely make automatic configuration easier.

**Comment by muesli** [comment]
We could solve that in a similar fashion as we only include some hives on particular platforms (`hives_linux.go` etc). I think it makes sense to move this to core, but maybe a systemd bee could still be cool: maybe we can trigger events when services are changing state, like (re)starting or stopping?

We could theoretically also start / stop services, even though we'd usually be limited to interacting with user services (we're not running as `root` after all).

**Comment by rubiojr** [comment]
Good ideas. I’ll start by abstracting the watchdog so we can add it to the core.

**Comment by rubiojr** [comment]
cc https://github.com/muesli/beehive/pull/275

**Comment by rubiojr** [comment]
https://github.com/muesli/beehive/pull/275 was merged so closing this.

**Pull request closed without merging**